File: src/main/java/ua/kostenko/recollector/app/dto/CategoryDto.java
package ua.kostenko.recollector.app.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CategoryDto {

    private Long categoryId;
    private String categoryName;
    private Long todoItems;
    private Long inProgressItems;
    private Long finishedItems;
}

File: frontend/src/api/dto/categoryDto.ts
import { FilterDirectionEnum } from "./common.ts";


export interface CategoryDto {
    categoryId?: number;
    categoryName?: string;
    todoItems?: number;
    inProgressItems?: number;
    finishedItems?: number;
}

export interface CategoryFilter {
    page?: number;
    size?: number;
    categoryName?: string;
    direction?: FilterDirectionEnum;
}


